code/game/objects/items/cigs_lighters.dm

```diff
@@ -157,12 +157,16 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	. = ..()
 
 /obj/item/clothing/mask/cigarette/attackby(obj/item/W, mob/user, params)
-	if(!lit && smoketime > 0)
-		var/lighting_text = W.ignition_effect(src, user)
-		if(lighting_text)
-			light(lighting_text)
-	else
+	if(lit || smoketime <= 0)
 		return ..()
+	if(!reagents.has_reagent(/datum/reagent/oxygen)) //cigarettes need oxygen
+		var/turf/open/location = get_turf(src)
```

Comment:
You really shouldn't be using the turf for these checks, for reasons laid out in the original issue. Check to see if loc isn't null, and use loc.return_air(). Or hell, call return_air() normally, it'll do the check for you.

Reply:
Initially when I first coded this, I tried to use loc to get the item location but it only worked if it was on the ground.  If the item was in a container or on the players mask slot, it would give me the other object instead of the turf.  This is why I used `get_turf`.  I will try it again with `return_air` and see if it will be any different.

Reply:
To be clear, this would only work if you put has_gas on gasmixtures rather then turfs

Reply:
Resolved with 73324af

Also the `return_air` suggestion was very helpful.  Thanks!